unix-domain-socket/src/main/scala/akka/stream/alpakka/unixdomainsocket/impl/UnixDomainSocketImpl.scala

```diff
@@ -256,12 +258,18 @@ private[unixdomainsocket] object UnixDomainSocketImpl {
         acceptedChannel.register(sel, SelectionKey.OP_READ, context)
       } catch { case _: IOException => }
       incomingConnectionQueue.offer(
-        IncomingConnection(localAddress, acceptingChannel.getRemoteSocketAddress, connectionFlow)
+        IncomingConnection(
+          UnixSocketAddress(Paths.get(localAddress.path())),
+          UnixSocketAddress(
+            Paths.get(Option(acceptingChannel.getRemoteSocketAddress).getOrElse(new JnrUnixSocketAddress("")).path())
+          ),
+          connectionFlow
+        )
```

Comment:
The instantiation of `IncomingConnection` may benefit from some named parameters in this expanded form.

Reply:
Will do.